### DISCUSSION TITLE
Cannot display H256 video with ui.video

### DISCUSSION THREAD

**Discussion by tiend1** [zauberzeug/nicegui#3154] [Q&A]
### Question

Note: Video has been removed because it showed inappropriate content.

**Comment by rodja** [comment]
It works for me. Could you share your code?

**Reply by tiend1** [reply]
Hi @rodja Thank for your reply. The code was taked from nicegui documentation.
```py
from nicegui import ui
import os

v = ui.video('1.mp4')
v.on('ended', lambda _: ui.notify('Video playback completed'))

ui.run()

```

**Reply by tiend1** [reply]
The version that i am using. nicegui:1.4.25

**Comment by rodja** [comment]
Could you also test, if the video plays in your browser directly (without any NiceGUI code)? The codec may not be supported in your specific configuration.

**Reply by tiend1** [reply]
Yes, The video is worked with vlc.
My orgin video is H265, I tried to convert it to H264, it work perfectly. But i don't want to use the converter, becase of high cpu usage.
converter.py
```py
from moviepy.editor import VideoFileClip

input_file = "1.mp4"  # Replace with your video filename
output_file = "converted_video.mp4"  # Replace with desired output

try:
    clip = VideoFileClip(input_file)
    clip.write_videofile(output_file, codec="libx264")
    print("Video conversion successful!")
except Exception as e:
    print(f"An error occurred during conversion: {e}")
```

Can you please check the video in [gdrive](https://drive.google.com/drive/folders/1Poj0rigqgKCCI0c0C9oFcyQgd57u0R-w)

**Reply by falkoschindler** [reply]
Besides running in VLC, does it run in the _browser_? You can try opening the file directly or use the following HTML snippet:
```html
<html>
  <body>
    <video width="320" height="240" controls>
      <source src="1.mp4" type="video/mp4" />
    </video>
  </body>
</html>
```

**Reply by tiend1** [reply]
Hi @falkoschindler Thank for your reply.

### On google chrome
With the h265 video:
![Screenshot from 2024-05-31 23-50-12](https://github.com/zauberzeug/nicegui/assets/166503915/5cf8897e-ebd4-4c16-ab86-56a7a19e6509)
With the h264 video (converted video)
![Screenshot from 2024-05-31 23-52-04](https://github.com/zauberzeug/nicegui/assets/166503915/64fdb9d5-ba65-443d-aad9-2c512b9c9b26)

### On VLC
Both of them work

**Comment by tiend1** [comment]
Please check the [reproduced video](https://drive.google.com/file/d/1JwCQ0NSrtHboG5Z9a4QHw32JUp1B5Sp7/view?usp=drive_link)

**Comment by tiend1** [comment]
The detail of the two video above

### H265 video
![Screenshot from 2024-05-31 23-54-05](https://github.com/zauberzeug/nicegui/assets/166503915/86d8512c-aa00-4937-a61a-612bf082c0b1)

### H264 video (converted video)
![Screenshot from 2024-05-31 23-54-20](https://github.com/zauberzeug/nicegui/assets/166503915/a80d22cc-a209-4de5-9ece-50794c0f7ebb)

**Comment by falkoschindler** [comment]
It seems to be a limitation of your browser and NiceGUI can't do anything about it.
A different browser might work. Or you need to install some codecs on your system. But I'm no expert in video compression on Linux.

**Reply by tiend1** [reply]
Thanks. Does the first H265 video (1.mp4) work on your browser?

**Reply by falkoschindler** [reply]
Yes, it does. (macOS, Chrome)
